File: front/src/store/audit.ts
import { ref } from 'vue';
import { defineStore } from 'pinia';
import { useWhereAmI } from '@/hooks/useWhereAmI';
import { IListResData, QueryBuilderType } from '@/typings';
import { enableCount } from '@/utils/search';
import http from '@/http';

export interface IAuditItem {
  id: number;
  res_id: string;
  cloud_res_id: string;
  res_name: string;
  res_type: string;
  associated_res_id: string;
  associated_cloud_res_id: string;
  associated_res_name: string;
  associated_res_type: string;
  action: string;
  bk_biz_id: number;
  vendor: string;
  account_id: string;
  operator: string;
  source: string;
  rid: string;
  app_code: string;
  detail: {
    data: any;
  };
  created_at: string;
  [key: string]: any;
}

export const useAuditStore = defineStore('audit', () => {
  const { getBusinessApiPath } = useWhereAmI();

  const isAuditListLoading = ref(false);
  const getAuditList = async (params: QueryBuilderType) => {
    isAuditListLoading.value = true;
    const api = `/api/v1/cloud/${getBusinessApiPath()}audits/list`;
    try {
      const [listRes, countRes] = await Promise.all<
        [Promise<IListResData<IAuditItem[]>>, Promise<IListResData<IAuditItem[]>>]
      >([http.post(api, enableCount(params, false)), http.post(api, enableCount(params, true))]);
      const [{ details: list = [] }, { count = 0 }] = [listRes?.data ?? {}, countRes?.data ?? {}];
      return { list, count };
    } catch (error) {
      console.error(error);
      return Promise.reject(error);
    } finally {
      isAuditListLoading.value = false;
    }
  };

  return {
    isAuditListLoading,
    getAuditList,
  };
});

File: pkg/api/core/audit/audit.go
/*
 * TencentBlueKing is pleased to support the open source community by making
 * 蓝鲸智云 - 混合云管理平台 (BlueKing - Hybrid Cloud Management System) available.
 * Copyright (C) 2022 THL A29 Limited,
 * a Tencent company. All rights reserved.
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at http://opensource.org/licenses/MIT
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on
 * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 * We undertake not to change the open source license (MIT license) applicable
 *
 * to the current version of the project delivered to anyone in the future.
 */

// Package audit ...
package audit

import (
	"hcm/pkg/api/data-service/cloud"
	"hcm/pkg/criteria/enumor"
	"hcm/pkg/dal/table/audit"
	"hcm/pkg/dal/table/cloud/load-balancer"
)

// Audit define audit.
type Audit struct {
	ID                   uint64                   `json:"id"`
	ResID                string                   `json:"res_id"`
	CloudResID           string                   `json:"cloud_res_id"`
	ResName              string                   `json:"res_name"`
	ResType              enumor.AuditResourceType `json:"res_type"`
	AssociatedResID      string                   `json:"associated_res_id"`
	AssociatedCloudResID string                   `json:"associated_cloud_res_id"`
	AssociatedResName    string                   `json:"associated_res_name"`
	AssociatedResType    string                   `json:"associated_res_type"`
	Action               enumor.AuditAction       `json:"action"`
	BkBizID              int64                    `json:"bk_biz_id"`
	Vendor               enumor.Vendor            `json:"vendor"`
	AccountID            string                   `json:"account_id"`
	Operator             string                   `json:"operator"`
	Source               enumor.RequestSourceType `json:"source"`
	Rid                  string                   `json:"rid"`
	AppCode              string                   `json:"app_code"`
	Detail               any                      `json:"detail,omitempty"` // Detail list接口该字段默认不返回
	CreatedAt            string                   `json:"created_at"`
}

// RawAudit define audit.
type RawAudit struct {
	ID                   uint64                   `json:"id"`
	ResID                string                   `json:"res_id"`
	CloudResID           string                   `json:"cloud_res_id"`
	ResName              string                   `json:"res_name"`
	ResType              enumor.AuditResourceType `json:"res_type"`
	AssociatedResID      string                   `json:"associated_res_id"`
	AssociatedCloudResID string                   `json:"associated_cloud_res_id"`
	AssociatedResName    string                   `json:"associated_res_name"`
	AssociatedResType    string                   `json:"associated_res_type"`
	Action               enumor.AuditAction       `json:"action"`
	BkBizID              int64                    `json:"bk_biz_id"`
	Vendor               enumor.Vendor            `json:"vendor"`
	AccountID            string                   `json:"account_id"`
	Operator             string                   `json:"operator"`
	Source               enumor.RequestSourceType `json:"source"`
	Rid                  string                   `json:"rid"`
	AppCode              string                   `json:"app_code"`
	Detail               *audit.BasicDetailRaw    `json:"detail,omitempty"` // Detail list接口该字段默认不返回
	CreatedAt            string                   `json:"created_at"`
}

// TargetGroupAsyncAuditDetail 目标组异步任务操作详情
type TargetGroupAsyncAuditDetail struct {
	LoadBalancer tablelb.LoadBalancerTable `json:"load_balancer"`
	ResFlow      *cloud.ResFlowLockReq     `json:"res_flow"`
}
